### PR TITLE
Add optional key messageKey to options param in register method

### PR DESCRIPTION
**Pull request by thelinuxkid**
Allows messages with different JSON keys to be processed when the app
is not in the foreground

### CODE REVIEW

**Comment by ajoyatgithub** [comment]
What does this do?

**Comment by thelinuxkid** [comment]
The key 'message' is not standard. For example, Mixpanel push messages use 'mp_message.' With the current implementation these messages would be ignored. See here: https://mixpanel.com/help/reference/android-push-notifications
android push notifications mixpanel mobile analytics

**Comment by ajoyatgithub** [comment]
Thanks. I get it now.

**Comment by blackjid** [comment]
:+1:

**Comment by blackjid** [comment]
maybe you could add the info to the Readme.md

**Comment by thelinuxkid** [comment]
Will do.

On Thu, Mar 26, 2015 at 2:09 PM, Juan Ignacio Donoso <
notifications@github.com> wrote:

> maybe you could add the info to the Readme.md
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/phonegap-build/PushPlugin/pull/464#issuecomment-86718834>
> .
>



-- 
Andres Buritica
[image: Twitter] <http://twitter.com/thelinuxkidd> [image: LinkedIn]
<http://www.linkedin.com/in/thelinuxkid> [image: about.me]
<http://about.me/thelinuxkid> [image: Blogger]
<http://blog.thelinuxkid.com/>  <https://github.com/thelinuxkid>

**Comment by manzapanza** [comment]
Hi @thelinuxkid! Do you was able to integrate the Mixpanel push notification using this plugin?
I applied your edits but I'm trying to understand what to do exactly with the `e.regid` because the documentation tells:

"Your GCM push server needs to know the regID before it can push to this device here is where you might want to send it the regID for later use." 

but I don't know how.

Do you use some others mixpanel plugins for cordova?

Thanks in advance!

**Comment by thelinuxkid** [comment]
I will add documentation and I will explain how to integrate with Mixpanel
plugin.

On Wednesday, July 15, 2015, Massimo Rangoni <notifications@github.com>
wrote:

> Hi @thelinuxkid <https://github.com/thelinuxkid>! Do you was able to
> integrate the Mixpanel push notification using this plugin?
> I applied your edits but I'm trying to understand what to do exactly with
> the e.regid because the documentation tells:
>
> "Your GCM push server needs to know the regID before it can push to this
> device here is where you might want to send it the regID for later use."
>
> but I don't know how.
>
> Do you use some others mixpanel plugins for cordova?
>
> Thanks in advance!
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/phonegap-build/PushPlugin/pull/464#issuecomment-121725959>
> .
>


-- 
Andres Buritica
[image: Twitter] <http://twitter.com/thelinuxkidd> [image: LinkedIn]
<http://www.linkedin.com/in/thelinuxkid> [image: about.me]
<http://about.me/thelinuxkid> [image: Blogger]
<http://blog.thelinuxkid.com/>  <https://github.com/thelinuxkid>

**Comment by peterbenoit** [comment]
HI @thelinuxkid I've been trying to get Mixpanel to work with this plugin for over a week now. Any documentation would be very helpful!

**Comment by thelinuxkid** [comment]
TL;DR use with this Mixpanel plugin
https://github.com/thelinuxkid/cordova-mixpanel. I will be providing
documentation for both later today.
On Wednesday, July 22, 2015, Peter Benoit <notifications@github.com> wrote:

> HI @thelinuxkid <https://github.com/thelinuxkid> I've been trying to get
> Mixpanel to work with this plugin for over a week now. Any documentation
> would be very helpful!
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/phonegap-build/PushPlugin/pull/464#issuecomment-123709217>
> .
>


-- 
Andres Buritica
[image: Twitter] <http://twitter.com/thelinuxkidd> [image: LinkedIn]
<http://www.linkedin.com/in/thelinuxkid> [image: about.me]
<http://about.me/thelinuxkid> [image: Blogger]
<http://blog.thelinuxkid.com/>  <https://github.com/thelinuxkid>

**Comment by thelinuxkid** [comment]
I made a small update to the README explaining how this change can be useful and pointed to an example. For those looking to integrate with Mixpanel, see the push notification section of https://github.com/thelinuxkid/cordova-mixpanel/blob/master/README.md for the setup.

**Comment by thelinuxkid** [comment]
Any chance of getting this merged?

**Comment by ajoyoommen** [comment]
Probably not anytime soon. Since this plugin is now deprecated and [phonegap-plugin-push](https://github.com/phonegap/phonegap-plugin-push) is officially supported.

**Comment by thelinuxkid** [comment]
Got it. Might need to patch that then. Thanks.

On Mon, Sep 14, 2015 at 9:42 PM, Ajoy Oommen notifications@github.com
wrote:

> Probably not anytime soon. Since this plugin is now deprecated and
> phonegap-plugin-push https://github.com/phonegap/phonegap-plugin-push
> is officially supported.
> 
> —
> Reply to this email directly or view it on GitHub
> https://github.com/phonegap-build/PushPlugin/pull/464#issuecomment-140278844
> .

## 

Andres Buritica
[image: Twitter] http://twitter.com/thelinuxkidd [image: LinkedIn]
http://www.linkedin.com/in/thelinuxkid [image: about.me]
http://about.me/thelinuxkid [image: Blogger]
http://blog.thelinuxkid.com/  https://github.com/thelinuxkid
